tests/testsSharedSlice.cpp

```diff
@@ -0,0 +1,1440 @@
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#include "tests-common.h"
+
+#include "gtest/gtest.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/SharedSlice.h>
+#include <velocypack/Slice.h>
+
+#include <memory>
+#include <variant>
+
+using namespace arangodb;
+using namespace arangodb::velocypack;
+
+void initTestCases(std::vector<Builder>& testCases) {
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::noneSlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::illegalSlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::nullSlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::trueSlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::falseSlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::zeroSlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::emptyStringSlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::emptyArraySlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::emptyObjectSlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::minKeySlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Slice::maxKeySlice());
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Value(42));
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.add(Value(-42));
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.openArray();
+    builder.add(Value(42));
+    builder.close();
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.openArray();
+    builder.add(Value("42"));
+    builder.close();
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.openObject();
+    builder.add("foo", Value(42));
+    builder.close();
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.openObject();
+    builder.add("bar", Value("42"));
+    builder.close();
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.addTagged(42, Value("42"));
+  }
+  {
+    auto& builder = testCases.emplace_back();
+    builder.addExternal(Slice::nullSliceData);
+  }
+  // TODO add something using an attributeTranslator
```

Comment:
Will it still be done?

Reply:
IMO the test suite is thorough enough. I just wanted to leave this as a note that this is not explicitly tested here.